src/pages/Profile/components/EditProfile.tsx

```diff
@@ -0,0 +1,155 @@
+  float: right;
+  width: 100;
+`;
+
+const ErrorContainer = styled.div`
+  color: "red";
+  font-size: 10px;
+`;
+const EditProfile = (props: {
+  profile: IUserRegister;
+  updateProfile: (values: IUserRegister) => void;
+}) => {
+  const { name, phone, email, city, state, country, address } = props?.profile;
+
+  let schema = yup.object().shape({
+    name: yup.string().required("Full name is required"),
+    phone: yup.number().required("Phone is required").positive().integer(),
+    email: yup.string().email("Valid Email is required"),
+    city: yup.string(),
+    state: yup.string(),
+    country: yup.string(),
+  });
+  const formik = useFormik({
+    initialValues: {
+      name: name || "",
+      email: email || "",
+      phone: phone || "",
+      city: city || "",
+      state: state || "",
+      country: country || "",
+      address: address || "",
+    },
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      props.updateProfile(values);
+    },
+    validateOnBlur: true,
+    validationSchema: schema,
+  });
+
+  return (
+    <FlexDiv style={{ marginLeft: "5%" }} vert align="flex-start">
+      <Text style={{ color: "blueviolet", margin: 20 }}> Profile Update </Text>
+      <FlexDiv align="center" justify="space-between">
+        <FlexDiv style={{ marginBottom: 20 }} vert>
+          <InputField
+            label="Name"
+            altTheme
+            name="name"
+            type="text"
+            onChange={formik.handleChange}
+            value={formik.values.name}
+          />
+          {formik.errors.email && (
+            <ErrorContainer>{formik.errors.name}</ErrorContainer>
+          )}
+        </FlexDiv>
+        <FlexDiv style={{ marginLeft: 15, marginBottom: 20 }} vert>
+          <InputField
+            label="Email"
+            altTheme
+            name="email"
+            type="email"
+            onChange={formik.handleChange}
+            value={formik.values.email}
+          />
+          {formik.errors.email && (
+            <ErrorContainer>{formik.errors.email}</ErrorContainer>
+          )}
+        </FlexDiv>
+      </FlexDiv>
+      <FlexDiv justify="space-between">
+        <FlexDiv style={{ marginBottom: 20 }} vert>
+          <InputField
+            label="Phone Number"
+            altTheme
+            name="phone"
+            type="text"
+            onChange={formik.handleChange}
+            value={formik.values.phone}
+          />
+          {formik.errors.phone && (
+            <ErrorContainer>{formik.errors.phone}</ErrorContainer>
+          )}
+        </FlexDiv>
+        <FlexDiv style={{ marginLeft: 15, marginBottom: 20 }} vert>
+          <InputField
+            label="City"
+            altTheme
+            name="city"
+            type="text"
+            onChange={formik.handleChange}
+            value={formik.values.city}
+          />
+        </FlexDiv>
+      </FlexDiv>
+      <FlexDiv style={{ marginBottom: 20 }} justify="space-between">
```

Comment:
We should add margin prop to styled component what do u think

Reply:
yeah we may need